software/30.Firebase/README.md

```diff
@@ -0,0 +1,206 @@
+# Workshop 30 - Firebase ✨
+
+✔️ Discover the power of an app development platform
+
+✔️ Create a collection in a NoSQL database
+
+✔️ Build an authentication system
+
+✔️ Get a preview of serverless functions    
+
+## Step 0 - Initialization
+
+All the required information to install the workshop's dependencies are given in the [SETUP.md](./SETUP.md)
+
+### Step 0.5 - Create your project
+Let's create your firebase project 🚀  
+For this, you should go to [the firebase console](https://console.firebase.google.com/) and choose a name for your project (you can use `firebase-workshop-poc` for example 😉)
+
+Firebase will then take you to a dashboard where you will be able to select an app on which you want to add Firebase:  
+![image](https://user-images.githubusercontent.com/49811529/193420648-40733ffd-a5f3-4935-915a-5c599d5476dd.png)
+
+Click on the web icon `</>`, you'll need to enter a name once again (using the same name as before is fine).
+> 💡 Don't select the hosting option, it's another great feature of Firebase but we won't cover it here
+
+And just with those few clicks we are done!  
+Firebase should give you a command to run in your local project to install the firebase package, as long as some starter code that you can paste in your `index.js` 🤩
+
+## Step 1 - Authentication
+Now we are ready to use our first Firebase service: Authentication 🔒  
+You just have to activate the `Email/Password` provider in the `Sign-in method` section of [your firebase console](https://console.firebase.google.com/):  
+![auth-console](https://user-images.githubusercontent.com/49811529/193460013-bda98391-0964-4481-a91f-e4029812ac2f.png)
+
+And you are ready to code!
+Let's write some code to create an account in `index.js` in a function with the following prototype:  
+```javascript
+function register(req, res) {
+    // Complete this code
+}
+
+app.post('/register', register);
+```
+> This function will be executed every time a `POST` request is made to `localhost:4000/register` 😉
+
+Your objective is to retrieve the `email` and `password` from the request body, create a user in Firebase, and return the registered user object on success (or an error message on failure).
+
+> 💡 You'll deal with [Promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise), so you'll need to use the [.then() and .catch() methods](https://www.freecodecamp.org/news/javascript-promise-methods/)
+ to return data, or [`async/await`](https://javascript.info/async-await) if you prefer
```

Comment:
Just `async/await` please, `.then` and `.catch` are messy...
Add an example of usage, people are not always fluent with async.